docs/website/docs/v1.x/dsl/http/index.md

```diff
@@ -1 +1,226 @@
+```
+
+### foldHttp
+
+Folds over the HTTP application by taking in two functions, one for failure and one for success respectively, and one
+more HTTP application.
+
+```scala
+  val app1 = Http.succeed("text1")
+  val app2 = Http.succeed("text2")
+  val app3 = app1.foldHttp(e => Http.fail(e), s => Http.succeed(s), app2)
+```
+
+## Http Combinators
+
+`Http` combinators are special operators that combine several HTTP applications into one. These are few handy combinators.
+
+### defaultWith
+
+Combines two HTTP applications into one. `++` is an alias for defaultWith. If the first HTTP application returns `None`
+the second HTTP application will be evaluated.
+
+```scala
+  val app1: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / "text" => Response.text("Hello World!")
+  }
+  val app2: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / "json" => Response.ok
+  }
+  val app = app1 ++ app2
+```
+
+### orElse
+
+Runs the first HTTP application but if it fails, runs other, ignoring the result from self. `<>` is an alias for orElse.
+
+```scala
+  val app = Http.fail(1) <> Http.succeed(2)
+```
+
+### andThen
+
+Runs the first HTTP application and pipes the output into the other. `>>>` is an alias for andThen.
+
+```scala
+  val app1 = Http.fromFunction[Int](a => a + 1)
+  val app2 = Http.fromFunction[Int](b => b * 2)
+  val app = app1 >>> (app2)
+```
+
+### compose
+
+Compose is similar to andThen. It runs the second HTTP application and pipes the output to the first HTTP
+application. `<<<` is the alias for compose.
+
+```scala
+  val app1 = Http.fromFunction[Int](a => a + 1)
+  val app2 = Http.fromFunction[Int](b => b * 2)
+  val app = app1 <<< (app2)
+```
+
+Apart from these, there are many more operators that let you transform an Http in specific ways.
+
+## HttpAppSyntax
+
+If the application is of type `Http[R, E, Request, Response]` i.e `HttpApp[R,E]`, there are some special operators
+available for them. Here are few handy `HttpApp` operators.
+
+### setMethod
+
+Overwrites the method in the incoming request to the `HttpApp`
+
+```scala
+  val app1: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / "text" => Response.text("Hello World!")
+  }
+  val app = app1 setMethod (Method.POST)
+```
+
+### patch
+
+Patches the response produced by the HTTP application using a `Patch`.
+
+```scala
+  val app1: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / "text" => Response.text("Hello World!")
+  }
+  val app = app1.patch(Patch.setStatus(Status.ACCEPTED))
+```
+
+### getBodyAsString
+
+`getBodyAsString` extract the body of response as a string and make it the output type.
+
+```scala
+  val app1: HttpApp[Any, Nothing] = Http.collect[Request] {
+    case Method.GET -> !! / "text" => Response.text("Hello World!")
+  }
+  val app: Http[Any, Throwable, Request, String] = app1.getBodyAsString
+```
```

Comment:
Add a section to cover Http to HttpApp using map, contramap, mapzio, contramapZIO, middleware etc.